metal/end_model/end_model.py

```diff
@@ -212,9 +212,6 @@ def train_model(self, train_data, dev_data=None, log_writer=None, **kwargs):
         # Convert input data to data loaders
         train_loader = self._create_data_loader(train_data, shuffle=True)
 
-        # Initialize the model
-        self.reset()
```

Comment:
Do we want to get rid of this totally?

Reply:
@bhancock8 presumably we want to keep this (if we thought it was useful to have in the first place), but just make sure it respects frozen params?

Reply:
See changes